akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -461,6 +461,12 @@ private[http] object HttpServerBluePrint {
       def emitErrorResponse(response: HttpResponse): Unit =
         emit(responseCtxOut, ResponseRenderingContext(response, closeRequested = true), () ⇒ completeStage())
 
+      def maybePullRequestParsingIn(): Unit =
```

Comment:
Yes, afaics this is correct. fwiw, we could also just remove `pullSuppressed` and use `!hasBeenPulled(requestParsingIn) && isAvailable(requestPrepOut)` instead.

I think the problem with that whole stage is that it is really hard to understand which states are mixed up there. I think at some point it needs a proper rewrite. Until then we try to only fix what's broken.